### ISSUE TITLE
Query with Range does not work - Mapping failure

### ISSUE THREAD

**Issue by skind** [hoonzis/fluentnest#45]
Thank you for this great library !

My concern is about the usage with a range query. When using this kind of query I am not able to
map my Field "ITC_DATE" because of the leading lower char! In Fiddler  I see this : "iTC_DATE".
This is here the problem, when using a term query it works! I assigned already the property with
the [Keyword(Name="ITC_DATE")], so it seems, that the Lib works different within the query types.
Here is a snippet of my query:  
.FilterOn( f => f.ITC_DATE == 1502435824363)  // this works, Unix-Timestamp, long-Value
.FilterOn( f => f.ITC_DATE >= startDate && f.ITC_DATE <= endDate )   // does NOT works, here I have the leadinglower character.
Please can you correct this behaviour - thank you in advance and best regards,
Simon

**Comment by filpen** [comment]
Thanks for your feedback, you are right, there is a missing case that is not handled in case of comparisons. I have written a red test to reproduce it in a branch. It shouldn't be hard to fix, it's probably just a matter of plugging ```GetFieldExpression``` in the ```GenerateComparisonFilter``` method. I am not sure when I'll be able to find time for it though, feel free to pick it up and make the test pass if you need it urgently.

**Comment by skind** [comment]
..very fast response - great. I will try to fix it on my downloaded code. Thank you for the hint
where to find the place in the code. I will give you feedback...
Thanks again and best regards,
Simon

**Comment by skind** [comment]
Yes, this was the reason. Fixed it locally - now the Field has the right value.

Another issue:
My request does not work, because in this case I have to use a **TermQuery**!  In another project
I used this type of query and it worked. So, is it possible to use this with this Library?  
Thank you for your help in advance!
Simon

**Comment by filpen** [comment]
Cool, feel free to send a PR if you fixed it so that it will be fixed for everyone else using it.

Term queries are abstracted in FluentNest so normally you won't need to use them directly, I suggest you to open a different issue with some more details and I'll try to help you there.

EDIT: I have created a gitter room at https://gitter.im/fluentnest/community

**Comment by skind** [comment]
I will try my best - I always have to work offline, because we are not allowd to connect to the internet because of security concerns.
Then I will open a new issue.
Thanks a lot.

**Comment by skind** [comment]
Hey filpen,
I cloned the project with Visual Studio - fixed the issue, commited it and tried to push the fix, but was not able to do this because of VS (?).., here is the output:
"master" wird per Push übertragen
Error: cannot spawn /c/program files (x86)/microsoft visual studio/2017/community/common7/ide/commonextensions/microsoft/teamfoundation/team explorer/Git/mingw32/libexec/git-core/git-askpass.exe: No such file or directory
Fehler beim Übertragen per Push an das Remoterepository: Git failed with a fatal error.
could not read Username for 'https://github.com': terminal prompts disabled
Pushing to https://github.com/hoonzis/fluentnest.git
Sorry, have no idea what is the reason and what to do ....

**Comment by filpen** [comment]
Thanks Simon, I really appreciate the effort :) please check out this guide on how to send a pull request: https://help.github.com/articles/creating-a-pull-request/

Practically, you can't directly push to this repository, you need to fork this repository on github, ideally create a branch where you fix the issue on your fork, and then create a pull request from your branch. If you have any questions feel free to ask either here or on gitter even though I understand you have limited connectivity so that might not be an option. In the meanwhile, vielen Dank im Voraus :)

**Comment by filpen** [comment]
Also, it might be easier for starting to use https://desktop.github.com/ instead of directly VS because you probably have not configured your GitHub credentials on your VS and I am not sure how to do it myself.
